### PR TITLE
Revert k8s-cloud-builder changes in kube-cross and enable variant building

### PR DESCRIPTION
**Pull request by justaugustus**
#### What type of PR is this?

/kind cleanup feature

#### What this PR does / why we need it:

**Unblocks releases**

(Selective revert of https://github.com/kubernetes/release/pull/1258, along with the below bits, following a convo with @cblecker and https://github.com/kubernetes/kubernetes/pull/90963#issuecomment-626504443.)

- images: Revert k8s-cloud-builder changes in kube-cross
- images/k8s-cloud-builder: Fix indentation
- kube-cross: Fixup version arguments
- kube-cross: Drop `SKOPEO_VERSION` from image config
- k8s-cloud-builder: Enable variant building
- gcb: Use k8s-cloud-builder image for stage/release artifact phases 
- images: Remove extraneous config values  
  - kube-cross: Drop `go1.12` variant
  - images: Don't use default `ARG` values in Dockerfiles

/assign @tpepper @cblecker 
/priority critical-urgent
cc: @kubernetes/release-engineering 
/hold for testing

#### Does this PR introduce a user-facing change?

```release-note
- images/k8s-cloud-builder: Support building against multiple kube-cross versions in parallel
- kube-cross: Build v1.14.2-4 and v1.13.9-5 images
```

### CODE REVIEW

File: images/build/cross/Dockerfile [hunk 1]
```diff
@@ -80,7 +62,7 @@ RUN echo "deb http://archive.ubuntu.com/ubuntu xenial main universe" > /etc/apt/
   && apt-get install -y build-essential \
   && for platform in ${KUBE_DYNAMIC_CROSSPLATFORMS}; do apt-get install -y crossbuild-essential-${platform}; done
 
-ARG PROTOBUF_VERSION=3.0.2
+ARG PROTOBUF_VERSION=3.9.1
```
Comment (cblecker) [inline images/build/cross/Dockerfile:65]: Is this this version it was before? Or should this be 3.0.2?

File: images/build/cross/Makefile [hunk 2]
```diff
@@ -20,13 +20,12 @@ IMAGE=kube-cross
 
 TAG?=$(shell git describe --tags --always --dirty)
 CONFIG?=go1.14
-KUBE_CROSS_VERSION?=v1.14.2-3
+KUBE_CROSS_VERSION?=v1.14.2-4
 
 # Build args
 GO_VERSION?=1.14.2
-PROTOBUF_VERSION?=3.11.4
+PROTOBUF_VERSION?=3.9.1
```
Comment (cblecker) [inline images/build/cross/Makefile:27]: Oh, is this what it was before, but now we're just updating the Dockerfile to match? Maybe the Dockerfile var should become a dummy value (in a follow up maybe) and just use this to override.

Comment (justaugustus) [reply images/build/cross/Makefile:27]: @cblecker -- Great point! Dropped in https://github.com/kubernetes/release/pull/1279/commits/2be1b2f6a7b738f6af7cb9ab99d8bbcb4e171b99.

**Comment by justaugustus** [comment]
Successful tests:

`k8s-cloud-builder`:
- 1.14: https://console.cloud.google.com/cloud-build/builds/18ea63f6-c01b-4343-9735-4421252f1de8?project=k8s-staging-releng
- 1.13: https://console.cloud.google.com/cloud-build/builds/65931ccc-8655-4a58-94e5-cb79221c63af?project=k8s-staging-releng

`kube-cross`:
- 1.14: https://console.cloud.google.com/cloud-build/builds/a49e3702-b7a8-407e-adfb-afcfa6ce9d4c?project=k8s-staging-build-image
- 1.13: https://console.cloud.google.com/cloud-build/builds/798d5fd4-7a23-409d-a889-2b57c5887a80?project=k8s-staging-build-image

I'm testing the staging `kube-cross:v1.13.9-5` image at https://github.com/kubernetes/kubernetes/pull/90963 (https://github.com/kubernetes/kubernetes/pull/90963/commits/600ef61fc3567bf17c44efbacd1124ae904bab88).

**Comment by justaugustus** [comment]
Successful `kube-cross` retest with `PROTOBUF_VERSION` bump removed:
- 1.14: https://console.cloud.google.com/cloud-build/builds/662fde6c-09fa-457f-8c25-4255968197bd?project=k8s-staging-build-image
- 1.13: https://console.cloud.google.com/cloud-build/builds/56a0283b-710d-4d4c-8a15-3cf14490a9c2?project=k8s-staging-build-image

File: images/build/cross/Dockerfile (hunk 1 above)
Comment (justaugustus) [reply images/build/cross/Dockerfile]: @cblecker -- Reverted that bump. I was attempting to bump protobuf to what's in k/k@`master`, but it doesn't look as simple as I assumed.

**Comment by justaugustus** [comment]
/hold cancel
(as I've addressed the comments)

**Comment by justaugustus** [comment]
Promotion PR here: https://github.com/kubernetes/k8s.io/pull/860

**Pull request merged**